vllm/entrypoints/disagg_connector.py

```diff
@@ -0,0 +1,234 @@
+import asyncio
+import json
+import signal
+import traceback
+import uuid
+# from fastapi.lifespan import Lifespan
+from asyncio import Queue
+from contextlib import asynccontextmanager
+from typing import AsyncGenerator
+
+import uvicorn
+import uvloop
+import zmq
+import zmq.asyncio
+from fastapi import FastAPI, Request
+from fastapi.responses import JSONResponse, StreamingResponse
+
+from vllm.logger import init_logger
+from vllm.utils import FlexibleArgumentParser
+
+# default prefill and decode addr
+time_out = 180
+fastapi_port = 8000
+prefill_addr = "ipc://localhost:7010"
+socket_prefill_num = 100
+decode_addr = "ipc://localhost:7020"
+socket_decode_num = 100
+context_type_json = "application/json"
+context_type_error = "error"
+
+# Cannot use __name__ (https://github.com/vllm-project/vllm/pull/4765)
+logger = init_logger('vllm.entrypoints.disagg_connector')
+
+
+@asynccontextmanager
+async def lifespan(app: FastAPI):
+    # create socket pool with prefill and decode
+    logger.info("start create_socket_pool")
+    app.state.zmqctx = zmq.asyncio.Context()
+    app.state.sockets_prefill = await create_socket_pool(
+        app.state.prefill_addr, socket_prefill_num, zmqctx=app.state.zmqctx)
+    logger.info("success create_socket_pool sockets_prefill")
+    app.state.sockets_decode = await create_socket_pool(
+        app.state.decode_addr, socket_decode_num, zmqctx=app.state.zmqctx)
+    logger.info("success create_socket_pool sockets_decode")
+    yield
+    ## close zmq context
+    logger.info("shutdown disagg connector")
+    logger.info("term zmqctx")
+    app.state.zmqctx.destroy(linger=0)
+
+
+app = FastAPI(lifespan=lifespan)
+
+
+# create async socket pool with num_sockets use ZMQ_DEALER
+async def create_socket_pool(url: str, num_sockets: int,
+                             zmqctx: zmq.asyncio.Context) -> Queue:
+    sockets: Queue[zmq.Socket] = Queue()
+    for i in range(num_sockets):
+        sock = zmqctx.socket(zmq.DEALER)
+        identity = f"worker-{i}-{uuid.uuid4()}"
+        sock.setsockopt(zmq.IDENTITY, identity.encode())
+        sock.connect(url)
+        logger.info("%s started at %s with queue size %s", identity, url,
+                    sockets.qsize())
+        await sockets.put(sock)
+    return sockets
+
+
+# select a socket and execute task
+async def execute_task_async(route: str, headers: dict, request: dict,
+                             sockets: Queue):
+    sock: zmq.Socket = await sockets.get()
+    try:
+        requestBody = json.dumps(request)
+        headersJson = json.dumps(headers)
+        logger.info("Sending requestBody: %s to %s with headers: %s",
+                    requestBody, route, headersJson)
+        await asyncio.wait_for(sock.send_multipart(
+            [route.encode(),
+             headersJson.encode(),
+             requestBody.encode()]),
+                               timeout=time_out)
+        logger.info("Sent end")
```

Comment:
We cannot have logs like this in production code.

These should be `logger.debug` at most (and probably removed).